lib/liam/common.rb

```diff
@@ -16,9 +16,7 @@ def client_options
     end
 
     def env_credentials
-      return @env_credentials if defined?(@env_credentials)
-
-      @env_credentials = credentials[ENV['RAILS_ENV']]
+      @env_credentials ||= credentials[ENV['RAILS_ENV']]
```

Comment:
I think this is the shorter case, but it doesn't assign nil values, e.g:

```ruby
ENVX = { 'RAILS_ENV' => 'staging' }

def credentials
  before = Time.now
  sleep 3
  p Time.now - before
  { 'stagingx' => 'holi' }
end

def env_credentials
  @env_credentials ||= credentials[ENVX['RAILS_ENV']]
end

def env_credentialx
  return @env_credentialx if defined?(@env_credentialx)

  @env_credentialx = credentials[ENV['RAILS_ENV']]
end

env_credentials # 3 seconds
env_credentials # 3 seconds

env_credentialx # 3 seconds
env_credentialx # no output
```

How much time could it take to load the credentials? I don't know

Reply:
We are handling the nil case already by failing early [here](https://github.com/archdaily/aws-liam/pull/15/files#diff-0ce417bee847d99dd35a5c6f3e4631dff848afb89562dc09e2919e4e515c87b3R65-R67).

Reply:
I think I understand your point. It could be faster when the variable is already defined, no matter what value it holds. However, I don't think that short-circuiting is going to get us a real-world benefit simply because this variable is not shared across instances, we don't have logic behind this, and also because we don't have `sleeps` laying around in the code haha.

Reply:
![tumblr_378d0a95dc9c5d5fff9089835e7f80a2_e28a88bd_400](https://user-images.githubusercontent.com/11888191/115405261-60028b80-a1ee-11eb-84ee-06a4570a4cc0.jpeg)